courses/views.py

```diff
@@ -1,3 +1,43 @@
-from django.shortcuts import render
+from django.views     import View
+from django.http      import JsonResponse
+from django.db.models import Count
 
-# Create your views here.
+from courses.models import Course
+
+class ListView(View):
+    def get(self, request):
+        category_id     = request.GET.get('category', None)
+        sub_category_id = request.GET.get('sub_category', None)
+        # 카테고리 분류
+        if sub_category_id:
+            course_list = Course.objects.filter(sub_category__id = sub_category_id)
+        elif category_id:
+            course_list = Course.objects.filter(sub_category__category_id = category_id)
+        else:
+            course_list = Course.objects.all()
+            
+        # 정렬
+        sort_name = request.GET.get('sort', None)
+        if sort_name:
+            my_dict = {
+                'lastest'     : '-created_at',
+                'satisfaction': '-review_count',
+                'likes'       : '-like_count'
+            }
+            course_list = course_list.annotate(review_count = Count('review'))
+            course_list = course_list.annotate(like_count = Count('liked_user'))
+            course_list.order_by(my_dict[sort_name])
+                
+        results = [
+            {
+                'id'          : course.id,
+                'title'       : course.title,
+                'sub_category': course.sub_category.name,
+                'user'        : course.user.korean_name,
+                'like'        : course.liked_user.count(),
```

Comment:
```suggestion
                'like'        : course.liked_user,
```

이렇게 확인해보시겠어요?
이미 count 를 진행했기 때문에 다시 count() 를 하지 않아도 괜찮아보입니다.

Reply:
아 넵 annotate로 만든 주석 필드(like_count)로 바꾸겠습니다.